.buildkite/dagster-buildkite/dagster_buildkite/package_build_spec.py

```diff
@@ -0,0 +1,247 @@
+import os
+from typing import Callable, List, Mapping, NamedTuple, Optional, Union
+
+from .python_version import AvailablePythonVersion
+from .step_builder import BuildkiteQueue
+from .steps.tox import build_tox_step
+from .utils import BuildkiteLeafStep, GroupStep
+
+_CORE_PACKAGES = [
+    "python_modules/dagster",
+    "python_modules/dagit",
+    "python_modules/dagster-graphql",
+    "js_modules/dagit",
+]
+
+
+def _infer_package_type(directory: str) -> str:
+    if directory in _CORE_PACKAGES:
+        return "core"
+    elif directory.startswith("examples/"):
+        return "example"
+    elif directory.startswith("python_modules/libraries/"):
+        return "extension"
+    else:
+        return "infrastructure"
```

Comment:
nit: maybe explicit `?` fall back instead of gear ? not terribly important

Reply:
Makes sense-- I actually think the gear matches the current packages it's applied to, but I can add an "unknown" case to disambiguate from "infrastructure".